CMakeLists.txt

```diff
@@ -25,13 +25,27 @@ project(opencensus-cpp VERSION 0.3.0 LANGUAGES CXX)
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(OPENCENSUS_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
+
+option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
+
+IF(MSVC)
+    add_definitions(-DNOMINMAX)
+ENDIF()
+
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
+    option(INSTALL_HEADERS "Install header files" ON)
+else()
+    option(INSTALL_HEADERS "Install header files" OFF)
+endif()
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+
+find_package(googletest REQUIRED) 
```

Comment:
CMake has a native module to find gtest:

https://cmake.org/cmake/help/latest/module/FindGTest.html

why not use it here?

Reply:
Ah, because you want to download it.

Reply:
Well, this is the rationale for changing from "inline" ExternalProject_Add to doing it via find_package
1. Adding install/export failed because I then also had to export the abseil targets, which seemed wrong to me
2. find_package is then a point of customization: in vcpkg I'll just delete the FindModule.cmake for abseil and googletest, and use the built-in packages for vcpkg
3. when abseil supports find_package directly (with a install/targets and a config-file) this approach will still work
4. for `GTest` the native module is missing gmock, unfortunately :( - else I would be happy to use that
5. the only thing that does not work properly is that I have to cmake build the abseil and googletest targets before I can build the opencensus. This is because (I think) ExternaProject_Add runs at build time (not configure time) and for some reason cmake is not smart enough to build the dependencies on demand. 
What I want is e.g. to build an opencensus test and automatically build abseil and gtest when they are required, while also supporting install/export for opencensus. I found an approach on stackoverflow, see the other reply. If this works, might that be okay? :)